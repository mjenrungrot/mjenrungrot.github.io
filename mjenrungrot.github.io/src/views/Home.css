main a:link {
  color: #1492ff;
  text-decoration: none;
}

main a:visited {
  color: #1492ff;
  text-decoration: none;
}

main a:hover {
  color: #1492ff;
  text-decoration: underline;
}

main a:active {
  color: #0075dc;
  text-decoration: underline;
}