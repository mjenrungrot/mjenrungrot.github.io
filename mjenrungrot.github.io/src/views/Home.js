// @flow

// eslint-disable-next-line no-unused-vars
import React from 'react';
// eslint-disable-next-line no-unused-vars
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';

import {withStyles} from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import Fade from '@material-ui/core/Fade';

import Main from '../layouts/Main';

const styles = (theme) => ({
  root: {
    ...theme.mixins.gutters(),
    paddingTop: theme.spacing(2),
    paddingBottom: theme.spacing(2),
  },
  paperFade: {
    in: true,
  },
});

/**
 * Home View Component
 * @param {object} props creation params
 * @return {object} Home component
 */
function Home(props) {
  const {classes} = props;
  return (
    <Main selectedTab="/">
      <Fade in>
        <Paper className={classes.root}>
          <Typography variant="h5" component="h3" gutterBottom>
            About this site
          </Typography>
          <Typography component="p" gutterBottom>
            Welcome to my personal website.
          </Typography>
          <Typography component="p" gutterBottom>
            Source available at{' '}
            <a href="https://github.com/mjenrungrot/mjenrungrot.github.io">
              Here
            </a>
            .
          </Typography>
        </Paper>
      </Fade>
    </Main>
  );
}

Home.propTypes = {
  classes: PropTypes.any.isRequired,
};

export default withStyles(styles)(Home);
