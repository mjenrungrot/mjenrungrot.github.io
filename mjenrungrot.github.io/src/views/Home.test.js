// eslint-disable-next-line no-unused-vars
import React from 'react';
import {shallow} from 'enzyme';
import Home from './Home';

describe('Home component', () => {
  test('renders', () => {
    const component = shallow(<Home />);
    expect(component).toMatchSnapshot();
  });
});
