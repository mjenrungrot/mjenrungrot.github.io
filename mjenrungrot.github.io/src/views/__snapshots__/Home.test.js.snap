// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders 1`] = `
<Home
  classes={
    Object {
      "paperFade": "Home-paperFade-2",
      "root": "Home-root-1",
    }
  }
/>
`;
